testonly/hammer/hammer.go

```diff
@@ -430,32 +430,46 @@ func (s *hammerState) retryOneOp(ctx context.Context) (err error) {
 	if s.chooseInvalid(ep, s.prng) {
 		glog.V(3).Infof("%d: perform invalid %s operation", s.cfg.MapID, ep)
 		invalidReqs.Inc(s.label(), string(ep))
-		return performOp(ctx, ep, s.prng, s.invalidReadOps, s.setLeavesInvalid)
+		op, err := getOp(ep, s.invalidReadOps, s.setLeavesInvalid)
+		if err != nil {
+			return err
+		}
+		return op(ctx, s.prng)
+	}
+
+	op, err := getOp(ep, s.validReadOps, s.setLeaves)
+	if err != nil {
+		return err
 	}
 
 	glog.V(3).Infof("%d: perform %s operation", s.cfg.MapID, ep)
-	defer func(start time.Time) {
-		rspLatency.Observe(time.Since(start).Seconds(), s.label(), string(ep))
-	}(time.Now())
 
-	deadline := time.Now().Add(s.cfg.OperationDeadline)
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	var firstErr error
+	return s.retryOp(ctx, op, string(ep))
+}
+
+func (s *hammerState) retryOp(ctx context.Context, fn mapOperationFn, opName string) error {
+	defer func(start time.Time) {
+		rspLatency.Observe(time.Since(start).Seconds(), s.label(), opName)
+	}(time.Now())
```

Comment:
What a nice hack (that somebody else gets the credit for :D).

Reply:
:-D